code/data_structures/src/list/singly_linked_list/operations/find/search_element_python.py

```diff
@@ -0,0 +1,73 @@
+class Node:
+    '''
+    A Node in a singly-linked list
+
+    Parameters
+    -------------------
+    data :
+        The data to be stored in the node
+    next:
+        The link to the next node in the singly-linked list
+    '''
+
+    def __init__(self, data=None, next=None):
+        self.data = data
+        self.next = next
+
+    def __repr__(self):
+        ''' Node representation as required'''
+        return (self.data)
+
+
+class SinglyLinkedList:
+    def __init__(self):
+        self.head = None
+
+    def append(self, data):
+        '''
+        Inserts New data at the ending of the Linked List
+        Takes O(n) time
+        '''
+        if not self.head:
+            self.head = Node(data=data)
+            return
+
+        current_node = self.head
+        while current_node.next:
+            current_node = current_node.next
+
+        new_node = Node(data=data, next=None)
+        current_node.next = new_node
+
+    def find(self, data):
+        '''
+        Returns the first index at which the element is found
+        Returns -1 if not found
+        Takes O(n) time
+        '''
+
+        if self.head is None:
```

Comment:
`return -1` even the list is empty.
Uses `if self.head is not None:` and remove the following return statement. code will be clearer.